pyiceberg/io/pyarrow.py

```diff
@@ -1912,3 +1920,55 @@ def _get_parquet_writer_kwargs(table_properties: Properties) -> Dict[str, Any]:
             default=TableProperties.PARQUET_PAGE_ROW_LIMIT_DEFAULT,
         ),
     }
+
+
+def _dataframe_to_data_files(
```

Comment:
I've moved this one to `pyarrow.py` since it is Arrow-specific. This also explains the local imports above to avoid circular imports.